### PR TITLE
[GCS]Reduce get operations of worker table

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
We test fault tolerance in a job with 3000+ actors. When an actor is restarted, we will restart all actors in order to avoid inconsistent state. GCS will receive a large number of WorkerFailure messages. It will first get and then put storage, which will exert great pressure on redis in a short time. The get operation is to check that whether the worker is exist. However, this is not likely to happen. As soon as the worker starts, it will register with GCS, so we delete the get operation.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
What's the worst case scenario in this case? Is worker entry overwritten by put? Is there any way we can avoid this behavior using Redis API?

**Comment by ffbin** [comment]
> What's the worst case scenario in this case? Is worker entry overwritten by put? Is there any way we can avoid this behavior using Redis API?

`NodeManager::HandleUnexpectedWorkerFailure` should guarantee the correctness of the processing logic.

**Comment by rkooo567** [comment]
Actually, @ffbin Why don't we just use `GETSET`? https://redis.io/commands/getset

This seems to be a safer choice (since we don't need to rely on correctness on other components that are not well tested).

**Comment by ffbin** [comment]
> GETSET

I can't provide a general GetSet interface in storage, because storage will support other distributed storage in addition to redis. I analyzed the core worker initialization logic and found that get judgment is redundant.

File: src/ray/gcs/gcs_server/gcs_worker_manager.cc [hunk 1]
```diff
@@ -38,44 +38,24 @@ void GcsWorkerManager::HandleReportWorkerFailure(
...
-      GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
+      RAY_CHECK_OK(gcs_pub_sub_->Publish(WORKER_CHANNEL, worker_id.Binary(),
+                                         worker_failure_data->SerializeAsString(),
+                                         nullptr));
     }
+    GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
   };
-  Status status = gcs_table_storage_->WorkerTable().Get(worker_id, on_get_done);
+
+  Status status =
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/gcs_worker_manager.cc:55]: Can you write a comment about this PR? That could be potentially useful if any issue happens in the future.

Comment (ffbin) [reply src/ray/gcs/gcs_server/gcs_worker_manager.cc:55]: Good idea, thanks

**Comment by rkooo567** [comment]
Lint failure unrelated. Can you make sure Java tests are fine and add test_ok tag when it is ready?

**Comment by ffbin** [comment]
Hi @rkooo567 , pls help take a look, thanks.

**Pull request merged**